src/z_data/serde/de.rs

```diff
@@ -143,10 +146,13 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
         V: Visitor<'de>,
     {
         match self.input {
-            ZData::Atom(x) if x.len() == 8 => {
-                let a: [u8; 8] = x.clone().try_into().unwrap();
-                visitor.visit_u64(u64::from_le_bytes(a))
-            }
+            ZData::Atom(x) => match x.clone().try_into() {
+                Ok(a) => visitor.visit_u64(u64::from_le_bytes(a)),
+                // This error can't happend?
+                Err(_) => Err(SerdeError::Type(
```

Comment:
This error could still happen if `x.len()` is not `8`, in which case the `.try_into()` would fail to convert it into a `[u8; 8]`.

This is because in your change you removed the `if x.len() == 8` condition in the match arm (before the `=>`).

Previously, in both this function (and the `deserialize_u32` function that has a similar pattern), it would never run the `x.clone().try_into().unwrap();` line unless the `if x.len() == 8` condition passed. This means that even though there was an `.unwrap()` here, it would never panic in practice, since the condition guarantees that the `.try_into()` should always succeed.

`cargo clippy` has no way of checking the above conditions, and that's why it listed this location among the rest.